packages/@aws-cdk/aws-ec2-alpha/README.md

```diff
@@ -366,3 +362,104 @@ myVpc.addInternetGateway({
+
+If you wish to add a new subnet to imported VPC, new subnet's IP range(IPv4) will be validated against provided secondary and primary address block to confirm that it is within the the range of VPC.
+
+Here's an example of how to import a VPC with multiple CIDR blocks, IPv6 support, and different subnet types:
+
+In this example, we're importing a VPC with:
+
+ - A primary CIDR block (10.1.0.0/16)
+ - One secondary IPv4 CIDR block (10.2.0.0/16)
+ - Two secondary address using IPAM pool (IPv4 and IPv6)
+ - VPC has Amazon-provided IPv6 CIDR enabled
+ - An isolated subnet in us-west-2a
+ - A public subnet in us-west-2b
+
+```ts
+
+const stack = new Stack();
+
+const importedVpc = VpcV2.fromVpcV2Attributes(this, 'ImportedVPC', {
+  vpcId: 'vpc-XXX',
+  vpcCidrBlock: '10.1.0.0/16',
+  secondaryCidrBlocks: [
+    {
+      cidrBlock: '10.2.0.0/16',
+      cidrBlockName: 'ImportedBlock1',
+    },
+    {
+      ipv6IpamPoolId: 'ipam-pool-XXX',
+      ipv6NetmaskLength: 52,
+      cidrBlockName: 'ImportedIpamIpv6',
+    },
+    {
+      ipv4IpamPoolId: 'ipam-pool-XXX',
+      ipv4IpamProvisionedCidrs: ['10.2.0.0/16'],
+      cidrBlockName: 'ImportedIpamIpv4',
+    },
+    {
+      amazonProvidedIpv6CidrBlock: true,
+    }
+  ],
+  isolatedSubnets: [{
+    subnetName: 'IsolatedSubnet2',
+    subnetId: 'subnet-03cd773c0fe08ed26',
+    subnetType: SubnetType.PRIVATE_ISOLATED,
+    availabilityZone: 'us-west-2a',
+    ipv4CidrBlock: '10.2.0.0/24',
+    routeTableId: 'rtb-0871c310f98da2cbb',
+  }],
+  publicSubnets: [{
+    subnetId: 'subnet-0fa477e01db27d820',
+    subnetType: SubnetType.PUBLIC,
+    availabilityZone: 'us-west-2b',
+    ipv4CidrBlock: '10.3.0.0/24',
+    routeTableId: 'rtb-014f3043098fe4b96',
+  }],
+});
+
+// You can now use the imported VPC in your stack
+
+// Adding a new subnet to the imported VPC
+const importedSubnet = new SubnetV2(this, 'NewSubnet', {
+  availabilityZone: 'us-west-2a',
+  ipv4CidrBlock: new IpCidr('10.2.2.0/24'),
+  vpc: importedVpc,
+  subnetType: SubnetType.PUBLIC,
+});
+
+// Adding gateways to the imported VPC
+importedVpc.addInternetGateway();
+importedVpc.addNatGateway({ subnet: importedSubnet });
+importedVpc.addEgressOnlyInternetGateway();
+```
+
+You can add more subnets as needed by including additional entries in the `isolatedSubnets`, `publicSubnets`, or other subnet type arrays (e.g., `privateSubnets`).
+
+### Importing Subnets
+
+You can also import individual subnets using the `SubnetV2.fromSubnetV2Attributes()` method. This is useful when you need to work with specific subnets independently of a VPC.
+
+Here's an example of how to import a subnet:
+
+```ts
+
+SubnetV2.fromSubnetV2Attributes(this, 'ImportedSubnet', {
+  subnetId: 'subnet-0123456789abcdef0',
+  availabilityZone: 'us-west-2a',
+  ipv4CidrBlock: '10.2.0.0/24',
+  routeTableId: 'rtb-0871c310f98da2cbb',
+  subnetType: SubnetType.PRIVATE_ISOLATED,
+});
```

Comment:
is there a use case that the customer needs to mention the VPC of this subnet ?

Reply:
I checked the implementation, we only need vpc info while creating a new subnet to validate fields like whether subnet range is within VPC range, or VPC supports IPv6, other than that it is not being referred anywhere so should be good without VPC information.